compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/DefaultArgumentStubGenerator.kt

```diff
@@ -109,15 +105,11 @@ open class DefaultArgumentStubGenerator(
             }
 
             when (irFunction) {
-                is IrConstructor -> +IrDelegatingConstructorCallImpl(
-                    startOffset = irFunction.startOffset,
-                    endOffset = irFunction.endOffset,
-                    type = context.irBuiltIns.unitType,
-                    symbol = irFunction.symbol,
-                    typeArgumentsCount = newIrFunction.parentAsClass.typeParameters.size + newIrFunction.typeParameters.size
-                ).apply {
+                is IrConstructor -> +irDelegatingConstructorCall(irFunction).apply {
                     passTypeArgumentsFrom(newIrFunction.parentAsClass)
-                    passTypeArgumentsFrom(newIrFunction)
+                    // This is for Kotlin/Native, which differs from the other backends in that constructors
+                    // apparently do have type parameters and dispatch receivers:
```

Comment:
Constructors can have their own type parameters in Java, but this is irrelevant in this case, since there are no default parameters in Java.
Have you seen constructors with type parameters in Native code? I never heard about those.

Reply:
Not sure about type parameters, actually, I've left this line in just in case. (There are definitely constructors with receivers, though.)

Reply:
Then I suggest leaving this line, as it makes no harm anyway, but removing the comment, because it seems misleading.

Reply:
Edited the comment to mention that there are probably no type parameters.